types/src/jsonrpc/wrapped/batches.rs

```diff
@@ -123,48 +119,50 @@ impl<T> BatchesState<T> {
 		// better performances in real-world situations. More brainstorming and benchmarking could
 		// get helpful here.
 
-		// Because of long-standing Rust lifetime issues
-		// (https://github.com/rust-lang/rust/issues/51526), we can't do this in an elegant way.
-		// If you're reading this code, know that it took several iterations and that I hated my
-		// life while trying to figure out how to make the compiler happy.
-		for batch_id in self.batches.keys().cloned().collect::<Vec<_>>() {
+		for (batch_id, entry) in self.batches.iter_mut().enumerate() {
```

Comment:
Is it true that "entry" is always a request? Can we name it "request"?

Reply:
`entry` is an option of a tuple of `(batch, user_param)`, not request per say. I reckon `entry` is a pretty recognizable term for a member of a collection/map.

Reply:
With the same logic we could name a variable `let int = some_call_that_returns_some_sort_of_integer()`, or `let string = get_input_from_keyboard()`. Variable names should strive to describe their purpose in the code, not what they are from a technical POV, i.e. have some kind of semantic value as opposed to just opaque markers. 
Here it's a bit tricky to come up with something good and terse, but given it comes from `self.batches` I think `batch` is decent enough (if you squint a bit).